# Lab 01 - Introduction to Experimentation

A reminder to **run the cell below** (by clicking on it and pressing the play/triangle button, or by Shift + Return) before getting started! This cell will import the necessary Python packages to run your data analysis in today's lab.

In [2]:
import numpy as np

Edit this cell and provide the information requested:

- Your name:
- Your student number:
- Partner's name:


## Part B - Familiarization with the experiment

*What is the main target quantity from the experimental goal?*
- We want to determine the spring constant, $k$, from a single measurement
*What can you control and what can you measure using the simulation?*
- We can control the degree to which we compress the spring (to a height $x_2$)
- We will measure the uncompressed spring mass ($m_1$) and height ($x_1$), and the compressed mass ($m_2$)
*How do you get from these measurable quantities to the target quantity?*
- We use the formula for Hooke's law, given below, with our measured quantities
- We will assume the local gravitational acceleration to be $g=9.81$ m/s^2

Initial measurement strategy:
- We will measure all masses using the electronic scale and heights with the ruler
- All measurements will be made with SI units (kg for mass, m for height)
- The heights will be measured from the top of the table (where the electronic scale is placed) to the end of the coil of the spring
    - Although this is measuring the height of the scale in addition to the height of the spring, since we only use $\Delta x$ in our calculation of $k$, this is OK
- We will compress the spring to roughly half of its initial height

General observations during measurement:
- The spring weighs about 78 g without any force applied
- The scale can only read to the nearest gram
- Pressing down on the spring and holding increases the scale reading to 100s or even 1000s of grams, depending on the force applied
- The compressed spring mass reading is not very stable

## Your initial measurement and calculation of $k$

Use the Python cell below to enter your measured values for $m_1$, $m_2$, $x_1$, and $x_2$. You may then calculate $k$ from these Python variables within the same cell.

In [3]:
# type in your measured values to the right of the '=' to define each Python variable below
m1 = 0.078 # uncompressed mass reading, in kg
x1 = 0.12 # uncompressed height, in m
m2 = 0.715 # compressed mass reading, in kg
x2 = 0.07 # compressed height, in m

# write some code below to calculate k
# it may be helpful to define intermediate variables, e.g. delta_m, to make your calculation easier to follow
g = 9.81 # acceleration due to gravity (in Vancouver), in m/s^2
dx = x2 - x1
dm = m2 - m1
dF = g * dm
k = np.abs(dF) / np.abs(dx)

print(k)

124.97940000000004


# Part C - Discussion with your group (Familiarization)

- Most other groups had spring constants on the order of 100 N/m (smallest: 98 N/m, largest 156 N/m)
- One group had a spring constant of about 1 N/m, but we realized they accidentally kept their $x_1$ and $x_2$ in cm
- Most other groups chose a compression of a few cm
- One group pressed the spring all the way to the scale, but their spring constant was much higher at 212 N/m, so there is probably a systematic error with that measurement technique

# Part D -  How the instructional team would perform the measurement

**Uncertainty in $m_1$:**
- Follows a uniform PDF
- Equal probability for all possible values in a range
- Parametrized by half-width, $a$
    - e.g., scale rounds to nearest gram so $a=0.5$ g
- Standard uncertainty is $\sigma = a/\sqrt{3}$

**Average:**
$$ \bar{x} = \frac{1}{N}\sum_{i=1}^N x_i $$
- $N$ is the number of measurements
- $x_i$ are the individual measurements
- For a Gaussian, the average characterizes the middle of the distribution (symmetric), i.e. the most likely value

**Standard deviation:**
$$ \sigma = \sqrt{\frac{1}{N-1} \sum_{i=1}^N (x_i - \bar{x})^2} $$
- Characterizes how far away (on average) a measurement is from the average of all measurements
- For a Gaussian, one standard deviation on either side of average corresponds to 68% of all values
- Two standard deviations = 95% of all values

**Uncertainty in $m_2$:**
- Fluctuating values (random) due to "shakiness" of hand
- Characterized by a Gaussian
- Four standard deviations = 95% confidence interval
- 95% CI is essentially the difference between the max and min observed values
- Standard uncertainty: $\sigma = 95\% CI / 4$

**Uncertainty in $x_1$ and $x_2$:**
- There are a number of small judgments needed to make these measurements (thus Gaussian)
- Use these factors to make a reasonable estimate for the 95% CI

# Part E - Revisiting your spring constant measurement with measurement uncertainty characterizations and calculations

## Notes from the investigation stage

- Since our measurement strategy worked well during familiarization, we will keep it the same as before

Uncertainty:
- $u[m_1]$ has a uniform PDF since it is a stable digital reading
- $u[m_2]$ has a Gaussian PDF since the value is digital, but fluctuating due to the "shakiness" of our hand
    - We will take the min and max observed mass reading during a five-second compression to calculate this uncertainty
- $u[x_1]$ and $u[x_2]$ will also have Gaussian PDFs, since factors such as slight tilts in the ruler, parallax, and the uncertainty on where exactly the top of the spring is provide random sources of uncertainty
    - We will eyeball the max and min plausible values off the ruler to calculate these uncertainties

## Measurement and data analysis

Add a new Python cell below and adapt your code from Part A to perform your improved measurement of $k$, along with uncertainty calculations. Consult the Appendix in the student instructions for helpful Python code which you can adapt to perform your uncertainty propagation calculations.

In [4]:
# type in your measured values to the right of the '=' to define each Python variable below
m1 = 0.078 # uncompressed mass reading, in kg
x1 = 0.12 # uncompressed height, in m
m2 = 0.845 # compressed mass reading, in kg
x2 = 0.06 # compressed height, in m

# write some code below to calculate k
# it may be helpful to define intermediate variables, e.g. delta_m, to make your calculation easier to follow
g = 9.81 # acceleration due to gravity (in Vancouver), in m/s^2
dx = x2 - x1
dm = m2 - m1
dF = g * dm
k = np.abs(dF) / np.abs(dx)

print(k)

# uncertainty calculations
um1 = 0.0005 / np.sqrt(3)
um2 = (0.855 - 0.835) / 4
ux1 = (0.125 - 0.115) / 4
ux2 = (0.065 - 0.055) / 4

udm = np.sqrt(um1**2 + um2**2)
udx = np.sqrt(ux1**2 + ux2**2)
udF = g * udm
uk = k * np.sqrt( (udF/dF)**2 + (udm/dm)**2 )

print(uk)

125.40450000000001
1.1580448501677312


# Part F - Summary of results (Start this by 2h30 after the start of the lab)

- Our spring constant is similar to our familiarization measurement, at $k=125.4 \pm 1.2$ N/m (compared to $125.0$ N/m from last time)
- The uncertainty seems reasonable, since it covers the difference between our two measurements
- Other groups had similar spring constants and uncertainty, such as $126.8 \pm 2.3$ N/m and $125.55 \pm 0.81$ N/m (uncertainties overlap with our value)
- One group had a similar value but wildly different uncertainty, $124 \pm 71$ N/m, which is likely a calculation error
- If we had more time, we would do even more measurements of $k$ and average the results
- We could also improve our measurement technique by using books to stabilize the compression to lower $u[m_2]$ and hence $u[k]$

# Submit your lab by 2h50m after the start of lab + 10-minute grace period

Steps for submission:

1. **Re-run all cells.** From the upper bar, select 'Kernel->Restart_Kernel_and_Run_All_Cells'
2. Look through the notebook to ensure all Markdown code formatted correctly and all Python code ran properly.
3. **Export to HTML.** Select 'File -> Save and Export Notebook As... -> HTML'.
4. Open up the HTML document and ensure everything looks correct.
5. Upload the HTML document to the lab submission assignment on Canvas.

You will receive a participation grade this week for a reasonable submission as well as feedback from a TA for next week.